lib/node_modules/@stdlib/blas/base/srotg/lib/assign.js

```diff
@@ -0,0 +1,91 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var abs = require( '@stdlib/math/base/special/abs' );
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var copysign = require( '@stdlib/math/base/special/copysign' );
+
+
+// MAIN //
+
+/**
+* Construct a Givens rotation from two values `a` and `b`.
+*
+* @private
+* @param {number} a - first value
+* @param {number} b - second value
+* @param {Float32Array} out - destination array
+* @param {number} stride - index increment
+* @param {number} offset - starting index
+* @returns {Float32Array} destination array
+*
+*
+* @example
+* var Float32Array = require( '@stdlib/array/float32' );
+* var out = srotg( 0.0, 2.0, new Float32Array( 4 ), 1, 0 );
+*  // returns <Float32Array>[ 2.0, 1.0, 0.0, 1.0 ]
+*/
+function srotg( a, b, out, stride, offset ) {
+	var scale;
+	var roe;
+	var r;
+	var c;
+	var s;
+	var z;
+
+	roe = b;
+	if ( abs( a ) > abs( b ) ) {
+		roe = a;
+	}
+	scale = abs( a ) + abs( b );
+	if ( scale === 0.0 ) {
+		c = 1.0;
+		s = 0.0;
+		r = 0.0;
+		z = 0.0;
+	} else {
+		r = scale * sqrt( pow( a/scale, 2 ) + pow( b/scale, 2 ) );
+		r = copysign( 1.0, roe ) * r;
+		c = a / r;
+		s = b / r;
+		z = 1.0;
+		if ( abs( a ) > abs( b ) ) {
+			z = s;
+		}
+		if ( abs( b ) >= abs( a ) && c !== 0.0 ) {
```

Comment:
I think you want `else if` here.

Reply:
https://github.com/xianyi/OpenBLAS/blob/develop/lapack-netlib/BLAS/SRC/srotg.f#L104 it had both IF statements, so...

Reply:
Both will be equivalent imo, will change to `else if`

Reply:
Actually, we should move L75 to L72 and do `else if ( aa > ab ) { z = s; }`. The second condition is more specific than the second.

Reply:
Correct. We're wasting cycles without the `else if`. As right now, L75 is always exercised atm even when L72 is true.